backend/battling/tests/test_view.py

```diff
@@ -0,0 +1,117 @@
+
+
+class CreateBattleViewTest(TestCase):
+    def setUp(self):
+        super().setUp()
+        self.client = Client()
+        self.creator = baker.make("users.User")
+        self.creator.set_password("admin")
+        self.creator.save()
+        self.opponent = baker.make("users.User")
+        self.opponent2 = baker.make("users.User")
+
+    def test_create_battle_with_user_not_logged(self):
+        battle_data = {
+            "creator": self.creator.id,
+            "opponent": self.opponent.email,
+        }
+
+        self.client.post("/battle/new/", battle_data)
+        battle = Battle.objects.filter(creator=self.creator, opponent=self.opponent)
+
+        self.assertFalse(battle)
+
+    def test_create_zero_battles(self):
+        self.client.login(username=self.creator.email, password="admin")
+        battle = Battle.objects.filter(creator=self.creator, opponent=self.opponent)
+        self.assertFalse(battle)
+
+    def test_create_one_battle(self):
+        battle_data = {
+            "creator": self.creator.id,
+            "opponent": self.opponent.email,
+        }
+
+        self.client.login(username=self.creator.email, password="admin")
+        self.client.post("/battle/new/", battle_data)
+        battle = Battle.objects.filter(creator=self.creator, opponent=self.opponent)
+
+        self.assertTrue(battle)
+
+        for _, single_battle in enumerate(battle):
+            self.assertCountEqual(single_battle.creator.email, self.creator.email)
+            self.assertCountEqual(single_battle.opponent.email, self.opponent.email)
+
+    def test_create_a_few_battles(self):
+        battle_data = {
+            "creator": self.creator.id,
+            "opponent": self.opponent.email,
+        }
+
+        self.client.login(username=self.creator.email, password="admin")
+        for _ in range(5):
+            self.client.post("/battle/new/", battle_data)
+
+        battle = Battle.objects.filter(creator=self.creator, opponent=self.opponent)
+        self.assertTrue(battle)
+
+        for _, single_battle in enumerate(battle):
+            self.assertCountEqual(single_battle.creator.email, self.creator.email)
+            self.assertCountEqual(single_battle.opponent.email, self.opponent.email)
+
+    def test_create_a_lot_of_battles(self):
+        battle_data = {
+            "creator": self.creator.id,
+            "opponent": self.opponent.email,
+        }
+
+        self.client.login(username=self.creator.email, password="admin")
+        for _ in range(50):
+            self.client.post("/battle/new/", battle_data)
+
+        battle = Battle.objects.filter(creator=self.creator, opponent=self.opponent)
+        self.assertTrue(battle)
+
+        for _, single_battle in enumerate(battle):
+            self.assertCountEqual(single_battle.creator.email, self.creator.email)
+            self.assertCountEqual(single_battle.opponent.email, self.opponent.email)
+
+    def test_create_battle_diff(self):
```

Comment:
The naming here could be improved. I suggest `test_create_battle_with_multiple_requests`